### PR TITLE
Remove logus for default logger and add windows support

### PR DESCRIPTION
**Pull request by doriac11**
# Description
- Add `//go:build !windows` in the drv_cfg.go so we can compile on windows.
- Remove references to `/sirupsen/logrus` for logging since it is no longer maintained. This was found during a blackduck scan. Instead use the go native logger: `https://pkg.go.dev/log/slog@go1.23.4` 

# Checklist:

- [x] I have performed a self-review of my own code to ensure there are no formatting, vetting, linting, or security issues
- [x] I have verified that new and existing unit tests pass locally with my changes
- [x] I have not allowed coverage numbers to degenerate
- [x] I have maintained at least 90% code coverage
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] Backward compatibility is not broken

### CODE REVIEW

**Review by rounak-adhikary — APPROVED** [review]

File: api.go [hunk 1]
```diff
@@ -282,7 +283,7 @@ func (c *Client) getStringWithRetry(
 	s, retry, httpErr := checkResponse(resp)
 	if httpErr != nil {
 		if retry {
-			doLog(log.Info, "need to re-auth")
+			logger.Info("need to re-auth")
```
Comment (shekhar-j) [inline api.go:286]: @doriac11 Can you wrap this in doLog?

File: api.go [hunk 2]
```diff
@@ -349,11 +350,10 @@ func NewClientWithArgs(
 		"debug":    debug,
 		"showHTTP": showHTTP,
 	}
-
-	doLog(log.WithFields(fields).Debug, "goscaleio client init")
+	doLog(logger.Debug, fmt.Sprintf("goscaleio client init, Fields: %+v", fields))
 
 	if endpoint == "" {
-		doLog(log.WithFields(fields).Error, "endpoint is required")
+		doLog(logger.Debug, fmt.Sprintf("endpoint is required, Fields: %+v", fields))
```
Comment (shekhar-j) [inline api.go:356]: this is .Error

**Review by shekhar-j — APPROVED** [review]

File: inttests/fs_test.go [hunk 3]
```diff
@@ -18,7 +18,6 @@ import (
 	"testing"
 
 	"github.com/dell/goscaleio"
-	// log "github.com/sirupsen/logrus"
```
Comment (JacobGros) [inline inttests/fs_test.go:21]: can we remove commented out line?

Comment (doriac11) [reply inttests/fs_test.go:21]: Hey @JacobGros , yes as part of this PR I removed the commented out references to `sirupsen/logrus`.

**Review by JacobGros — APPROVED** [review]

**Pull request merged**